feat: make transaction subscription configurable via CLI flag

sumitvekariya opened the pull request:
Similar to how pre-confirmation subscription is configurable, this change makes transaction subscription configurable with a new `--subscribe-to-transactions` CLI flag that defaults to false.

This allows nodes to opt out of receiving transaction gossip messages when they don't need them, reducing network traffic.

## Linked Issues/PRs
<!-- List of related issues/PRs -->
#2970

## Description
This PR introduces the following changes:

- Added a new CLI flag `--subscribe-to-transactions` that defaults to `false`
- Updated the P2P configuration to pass this flag to the Config object
- Modified the Gossipsub initialization logic to respect this flag when subscribing to transaction topics
- Made transaction subscription work the same way as pre-confirmation subscription

These changes help reduce network traffic for nodes that don't need to receive transaction gossiping.

## Checklist
- [x] Breaking changes are clearly marked as such in the PR description and changelog
- [x] New behavior is reflected in tests
- [x] [The specification](https://github.com/FuelLabs/fuel-specs/) matches the implemented behavior (link update PR if changes are needed)

### Before requesting review
- [x] I have reviewed the code myself
- [x] I have created follow-up issues caused by this PR and linked them here

### After merging, notify other teams

[Add or remove entries as needed]

- [ ] [Rust SDK](https://github.com/FuelLabs/fuels-rs/)
- [ ] [Sway compiler](https://github.com/FuelLabs/sway/)
- [ ] [Platform documentation](https://github.com/FuelLabs/devrel-requests/issues/new?assignees=&labels=new+request&projects=&template=NEW-REQUEST.yml&title=%5BRequest%5D%3A+) (for out-of-organization contributors, the person merging the PR will do this)
- [ ] Someone else?

xgreenx commented on crates/services/p2p/src/config.rs:
```diff
@@ -254,6 +258,7 @@ impl Config<NotInitialized> {
             tx_pool_threads: 0,
             state: NotInitialized,
             subscribe_to_pre_confirmations: true,
+            subscribe_to_transactions: false,
```
We want to enable it by defautl for tests

```suggestion
            subscribe_to_transactions: true,
```

xgreenx reviewed (commented):
It would be nice if you added an integration test that starts 2 nodes(producer and validator) to ensure that transctions are not broadcasted.

sumitvekariya commented:
> It would be nice if you added an integration test that starts 2 nodes(producer and validator) to ensure that transctions are not broadcasted.

Hey, I have addressed your feedback in recent commits, please check

xgreenx commented on tests/tests/tx_subscription.rs:
```diff
@@ -0,0 +1,448 @@
+        .next()
+        .expect("Producer is initialized with one transaction");
+
+    // Wait for the transaction to be committed on the producer node
+    let _ = client_one.await_transaction_commit(&tx_id).await.unwrap();
+
+    // Give some time for the transaction to be gossiped
+    tokio::time::sleep(Duration::from_secs(2)).await;
+
+    // Try to subscribe to the transaction on the validator node, this should work since gossip is enabled by default
+    let mut client_two_subscription = client_two
+        .subscribe_transaction_status(&tx_id)
+        .await
+        .expect("Should be able to subscribe for events");
+
+    // The transaction should be available on the validator node
+    tokio::time::timeout(Duration::from_secs(10), client_two_subscription.next())
+        .await
+        .expect("Should await transaction notification in time");
+
+    let response = client_two.transaction(&tx_id).await.unwrap();
+    assert!(
+        response.is_some(),
+        "Transaction should have been gossiped to validator"
+    );
```
You can use `client_two` to commit any transaction to it, if gossiping is enabled, `client_one`(a.k.a. block prodcuer) will include it.

It will be a proof that gossiping enabled.

xgreenx reviewed (changes requested):
Next tests are the same and don't test that they suppose to: `test_gossipsub_subscription_respects_flag`, `test_tx_subscription_flag_is_applied`, `test_tx_gossiping_can_be_disabled`

sumitvekariya commented:
> Next tests are the same and don't test that they suppose to: `test_gossipsub_subscription_respects_flag`, `test_tx_subscription_flag_is_applied`, `test_tx_gossiping_can_be_disabled`

please check the latest commit with updated tests addressing your feedback

Dentosal commented:
Closing in favour of #3083 which implements requested changes.
